addons/web_unsplash/static/src/js/unsplash_image_widget.js

```diff
@@ -130,56 +104,56 @@ widgetsMedia.ImageWidget.include({
      * @override
      */
     _highlightSelected: function () {
-        var self = this;
-        if (!this._unsplash.isActive) {
-            return this._super.apply(this, arguments);
-        }
+        this._super.apply(this, arguments);
 
-        this.$('.o_unsplash_img_cell.o_selected').removeClass('o_selected');
-        var $select = this.$('.o_unsplash_img_cell [data-imgid]').filter(function () {
-            return $(this).data('imgid') in self._unsplash.selectedImages;
-        });
-        $select.closest('.o_unsplash_img_cell').addClass('o_selected');
+        this.$('.o_unsplash_attachment_cell.o_we_attachment_selected').removeClass('o_we_attachment_selected');
+        const $select = this.$('.o_unsplash_attachment_cell[data-imgid]').filter((i, el) => {
+            return $(el).data('imgid') in this._unsplash.selectedImages;
+        }).addClass('o_we_attachment_selected');
         return $select;
     },
     /**
      * @private
      */
     _loadMoreImages: function (forceSearch) {
-        this._super(this._unsplash.isActive || forceSearch);
+        this._super(this.$('.o_we_search').val() !== '');
     },
     /**
      * @override
      */
     _renderImages: function () {
-        if (!this._unsplash.isActive) {
-            return this._super.apply(this, arguments);
-        }
-
+        this._super(...arguments);
         if (this._unsplash.error) {
-            this.$('.unsplash_img_container').html(
+            this.$('.unsplash_error').html(
                 core.qweb.render('web_unsplash.dialog.error.content', {
                     status: this._unsplash.error,
                 })
             );
             return;
         }
 
-        this.$('.unsplash_img_container').html(core.qweb.render('web_unsplash.dialog.image.content', {records: this._unsplash.records}));
-        this._highlightSelected();
-
-        this.$('.o_load_more').toggleClass('d-none', !!this._unsplash.error || this._unsplash.isMaxed);
-        this.$('.o_load_done_msg').toggleClass('d-none', !!this._unsplash.error || !this._unsplash.isMaxed);
+        if (!this._unsplash.isMaxed) {
+            this.$('.o_load_more').removeClass('d-none');
+            this.$('.o_load_done_msg').addClass('d-none');
+        }
     },
     /**
-     * @private
-     * @param {boolean} show
+     * @override
      */
-    _toggleUnsplashContainer: function (show) {
-        this._unsplash.isActive = show;
-        this.$('.o_we_existing_attachments').toggleClass('d-none', show);
-        this.$('.unsplash_img_container').toggleClass('d-none', !show);
-        this.$('.o_we_search_icon > span').text(show ? "Unsplash" : "");
+    _renderExisting: function (attachments) {
+        if (this._unsplash.records.length) {
+            attachments.splice(MAX_DB_ATTACHMENTS);
+        }
+        return this._super([...attachments, ...this._unsplash.records]);
```

Comment:
Original way to write `this._super(attachments.concat(this._unsplash.records))`, I am fine with both. Pretty sure the `concat` one is more readable if not more efficient though ;)